cinder/api/metricutil.py

```
@@ -4,9 +4,110 @@
 @author: souvik
 '''
 from metrics.ThreadLocalMetrics import ThreadLocalMetrics, ThreadLocalMetricsFactory
+from metrics.Metrics import Unit
 from oslo_log import log as logging
+from time import time
+from oslo_context.context import RequestContext as context
 LOG = logging.getLogger(__name__)
 
+class ReportMetrics(object):
+    def __init__(self, metric_name, report_error = False):
+        self.__metric_name = metric_name
+        self.__report_error = report_error
+    def __call__(self, function):
+        def metrics_wrapper(*args, **kwargs):
+            start_time = time()
+            error = 0
+            try:
+                return function(*args, **kwargs)
+            except Exception as e:
+                LOG.exception('Exception in Report Metrics: %s', e)
```

IIUC, this isn't an exception in report metrics; rather an exception in the actual function call itself...
So, the message needs to change.
Does LOG.exception reraise the exception? If not, the exception would be silently ignored. So, we need to reraise it.